src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -89,23 +95,28 @@ private void Initialize(Compilation comp, AnalyzerExecutor analyzerExecutor, Can
                     _compilationEndActionsMap = MakeCompilationActionsByAnalyzer(this.analyzerActions.CompilationEndActions);
                 }, cancellationToken, TaskContinuationOptions.None, TaskScheduler.Default);
 
-                // create the primary driver task.
                 cancellationToken.ThrowIfCancellationRequested();
-                _primaryTask = Task.Run(async () =>
+
+                // Start the initialize task.
+                _initializeTask = Task.Run(async () =>
```

Comment:
Why do we need to wrap initializeTask within another task? GetAnalyzerActionsAsync already returns a task, can't we just assign remove the local initializeTask and replace it with field _initializeTask?